src/zenml/integrations/kubeflow/flavors/kubeflow_orchestrator_flavor.py

```diff
@@ -116,6 +121,37 @@ def _migrate_pod_settings(cls, values: Dict[str, Any]) -> Dict[str, Any]:
             values["node_affinity"] = {}
             values["node_selectors"] = {}
 
+        client_creds_error = AssertionError(
+            "`client_username` and `client_password` both need to be set."
+        )
+        if (
+            values.get("client_username")
+            and values.get("client_password") is None
+        ):
+            raise client_creds_error
+        if (
+            values.get("client_password")
+            and values.get("client_username") is None
+        ):
+            raise client_creds_error
+
+        if values.get("kubeflow_hostname"):
+            if "/pipeline" not in values.get("kubeflow_hostname"):
```

Comment:
This should probably also be `hostname.endswith("/pipeline")`?

Reply:
Yeah but it might also be `/pipeline/` so thats why I just made it easier. WDYT?

Reply:
Good point, makes sense 👍